var/spack/repos/builtin/packages/nvidia-cuda-samples/package.py

```diff
@@ -0,0 +1,60 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import shutil
+
+from spack.package import *
+
+
+class NvidiaCudaSamples(CMakePackage, MakefilePackage, CudaPackage):
+    """A collection of NVIDIA curated sample applications showcasing features of CUDA"""
+
+    homepage = "https://github.com/NVIDIA/cuda-samples"
+    url = "https://github.com/NVIDIA/cuda-samples"
```

Comment:
The following will help you switch to sha256s:

```suggestion
    url = "https://github.com/NVIDIA/cuda-samples/archive/refs/tags/v12.8.tar.gz"
```

With that and the `versions` change, you can run `spack checksum nvidia-cuda-samples` and get the sha256s for each listed version.